extension/src/popup/basics/Toggle/index.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { cloneElement, useEffect, useState } from "react";
+import "./styles.scss";
+
+enum LabelPosition {
+  left = "left",
+  right = "right",
+}
+
+interface ToggleComponent {
+  labelPosition: typeof LabelPosition;
+}
+
+interface ToggleProps extends React.InputHTMLAttributes<HTMLInputElement> {
+  id: string;
+  checked: boolean;
+  customInput?: React.ReactElement;
+  onChange?: () => void;
+  disabled?: boolean;
+  labelOn?: string;
+  labelOff?: string;
+  labelPosition?: LabelPosition;
+}
+
+export const Toggle: React.FC<ToggleProps> & ToggleComponent = ({
+  id,
+  checked,
+  customInput,
+  onChange,
+  disabled,
+  labelOn,
+  labelOff,
+  labelPosition = LabelPosition.left,
+}: ToggleProps) => {
+  const [checkedValue, setCheckedValue] = useState(checked);
+
+  useEffect(() => {
+    setCheckedValue(checked);
+  }, [checked]);
+
+  const additionalClasses = [...(disabled ? ["Toggle--disabled"] : [])].join(
+    " ",
+  );
+
+  const renderLabel = () => (
+    <span>{checked ? labelOn : labelOff ?? labelOn}</span>
```

Comment:
I also made a small change here where you only have to pass it `labelOn` if you don't want the label to change on state change

Reply:
This is fine if we want to have the label unchanged between the states.

I just noticed that I made a mistake in not rendering the `span` conditionally if there isn't `labelOn` or `labelOff`. Right now it will always render a `span` there. I'll make sure we update it when we add this change to SDS.